server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -478,4 +485,38 @@ public void clear() {
             runnables.clear();
         }
     }
+
+    private boolean canMatch(LeafReaderContext ctx) throws IOException {
+        // skip segments for search after if min/max of them doesn't qualify competitive
+        return canMatchSearchAfter(ctx);
```

Comment:
Curios why do we need the `canMatchSearchAfter` instead of just doing the work in  `canMatch`? The  `canMatchSearchAfter`  is not used anywhere here (except this delegation), asaikt

Reply:
I am planning to add canMatchRange soon, which will be short cutting segments based on range queries. 
Initially I thoguht of having some canMatch interface and canMatchSearchAfter / canMatchRange etc canMatch shortcutting phase can implement this interface and we can simply call it as a list without worrying about impementation and each canMatch can have their own implementation (unit testable).

but later thought individual methods can do this job in simple way here as the implementations itself are not big enough. 
So I thoguht of having canMatch as a single functon top level module to be calling individual canMatch like search_after in this case......

More of readibility purpose and easy to debug if some bug.